tests/integration/test.basics.js

```diff
@@ -1043,6 +1043,10 @@ adapters.forEach(function (adapter) {
       db.type().should.be.a('string');
     });
 
+    it('#4788 db.type() is synchronous', function () {
+      new PouchDB(dbs.name).type.should.be.a('function');
```

Comment:
Unless I'm mistaken, this test will succeed both before and after the fix, so it doesn't catch it. Could you duplicate line 1047, so that the same thing is called twice (as in your example)? IME that reproduced the bug.